### PR TITLE
Update configure and bootstrap, fix compiler warnings in http_endpoin…

### PR DESCRIPTION
**Pull request by beneschtech**
### Issue or RFC Endorsed by Maintainers

None

### Description of the Change

Updated some of the build scripts to be more portable and to build on FreeBSD.
Fixed a few compiler warnings
Added OS specific readmes for applicable issues to help users

### Possible Drawbacks

Potentially break other platforms

### Verification Process

Built on FreeBSD 12, Fedora 32, CentOS 7, and Mingw64 (Windows 10)

### Release Notes

Added OS specific tips, and cleaned up some compiler warnings

### CODE REVIEW

File: src/httpserver/details/http_endpoint.hpp [hunk 1]
```diff
@@ -120,12 +120,12 @@ class http_endpoint
...
         }
 
-        const bool is_family_url() const
+        const bool is_family_url()
         {
             return family_url;
         }
 
-        const bool is_regex_compiled() const
+        const bool is_regex_compiled()
```
Comment (etr) [inline src/httpserver/details/http_endpoint.hpp:128]: what was the issue with the is_family_url and is_regex_compiled methods being const?

**Comment by beneschtech** [comment]
Compiler warning. It has const before it still.

**Comment by etr** [comment]
Got it. I think the real issue is the "const bool" (which makes no sense).

Does using "bool is_regex_compiled() const" cause you any warning?

Effectively we want the method to be declared as not modifying the object so I am not sure what the warning against that is.

**Comment by beneschtech** [comment]
As is:
In file included from
../lib/include/httpserver/webserver.hpp:46:../lib/include/httpserver/de
tails/http_endpoint.hpp:123:9: warning: 'const' type qualifier on
return type has no effect [-Wignored-qualifiers]        const bool
is_family_url()
const        ^~~~~~../lib/include/httpserver/details/http_endpoint.hpp:
128:9: warning: 'const' type qualifier on return type has no effect [-
Wignored-qualifiers]        const bool is_regex_compiled() const
Clang complains about the const in front, gcc is okay with either one
removed.
Just pushed new code with const at the end
On Fri, 2020-08-07 at 22:01 -0700, Sebastiano Merlino wrote:
> Got it. I think the real issue is the "const bool" (which makes no
> sense).
> Does using "bool is_regex_compiled() const" cause you any warning?
> Effectively we want the method to be declared as not modifying the
> object so I am not sure what the warning against that is.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
-- 
Chris Benesch <chris@beneschtech.com>

**Comment by etr** [comment]
Thanks!
All travis/appveyor tests are passing and all looks good. Thanks a lot for the help.

**Pull request merged**